docs/datascientist_onboard.md

```diff
@@ -0,0 +1,230 @@
+        ```
+    Reference: https://www.alibabacloud.com/help/zh/doc-detail/53928.htm
+ 
+- For Pytorch or other DL framworks in Python
+
+    There is no complete instance on the Internet for pytorch to solve the hdfs small file problem, but users can refer to the following method to build a custom dataloader to try to solve this problem. Official opinion on "pytorch reading hdfs": [issue-97](https://github.com/chainer/chainermn/issues/97), [issue-5867](https://github.com/pytorch/pytorch/issues/5867)
+
+    (1) Create [sequence file](https://wiki.apache.org/hadoop/SequenceFile) at HDFS
+  
+    Reference example scripts to write a sequence file: [SequenceFileWriterDemo.py](https://github.com/matteobertozzi/Hadoop/blob/master/python-hadoop/examples/SequenceFileWriterDemo.py)
+  
+  (2) Build customize Pytorch data reader to read from HDFS sequence file
+  
+    - (2.1) Build Pytorch [customize dataloader](https://discuss.pytorch.org/t/loading-huge-data-functionality/346/2)
+    
+    Just define a Dataset object, that only loads a list of files in __init__, and loads them every time __getindex__ is called. Then, wrap it in a torch.utils.DataLoader with multiple workers, and you’ll have your files loaded lazily in parallel.
+
+        ``` bash
+        class MyDataset(torch.utils.Dataset):
+            def __init__(self):
+                self.data_files = os.listdir('data_dir')
+                sort(self.data_files)
+            def __getindex__(self, idx):
+                return load_file(self.data_files[idx])
+            def __len__(self):
+                return len(self.data_files)
+        dset = MyDataset()
+        loader = torch.utils.DataLoader(dset, num_workers=8)
+        ``` 
+     - (2.2) Use python [sequence file reader](https://github.com/matteobertozzi/Hadoop/blob/master/python-hadoop/examples/SequenceFileReader.py) method rewrite related methods of dataloader 
+
+
+### 2. Prepare your job container
+PAI run jobs in containers, it supports all the DL frameworks (including CNTK, TensorFlow, etc.) by installing the corresponding packages in your customized job container.
+
+To prepare your customized job container, you should have a [Docker installed](https://docs.docker.com/install/linux/docker-ce/ubuntu/) machine, and a Docker Registry to store your containers, for any node of PAI can pull and get access to the containers. You can use the public  [Docker Hub](https://hub.docker.com/) or any other private registry.
+
+For your convenience, we have pre-built some images for you:
+  - openpai/pai.example.tensorflow
+  - openpai/pai.example.tensorflow-serving
+  - openpai/pai.example.keras.tensorflow
+  - openpai/pai.example.sklearn
+  - openpai/pai.example.pytorch
+  - openpai/pai.example.kafka
+  - openpai/pai.example.mxnet
+  - openpai/pai.example.caffe2
+You can find their Dockerfile in the corresponding directory of [PAI job examples](https://github.com/Microsoft/pai/tree/master/examples).
```

Comment:
I think we could add some tutorial here to teach scientists how to debug their container locally, avoiding a bad usage experience at the beginning, although we can't hide the docker details completely. 

Eg: Jobs may fail due to some little bugs(typo, path errors), and it would take scientists much time to fix them (finding bugs, fixing, rebuilding, pushing, pulling), especially in the case of users packing data into container for convenience. In my experience, testing such containers locally before submitting is very helpful. If `docker run -v ...:codeDir -v ...:dataDir -v ...:outputDir image_name /bin/bash -c command ` works well locally, then such job should success in PAI as well.